src/lib.rs

```diff
@@ -1061,17 +1065,177 @@ fn digraph_astar_shortest_path(
     Ok(out_path.to_object(py))
 }
 
-/// The provided node is invalid.
+/// Return a :math:`G_{np}` directed random graph, also known as an
+/// Erdős-Rényi graph or a binomial graph.
+///
+/// The :math:`G_{n,p}` graph algorithm chooses each of the
+/// :math:`n (n - 1)` possible edges with probability :math:`p`.
+/// This algorithm [1]_ runs in :math:`O(n + m)` time, where :math:`m` is the
+/// expected number of edges, which equals :math:`p n (n - 1)/2`.
+///
+/// Based on the implementation of the networkx function
+/// ``fast_gnp_random_graph`` [2]_
+///
+/// :param int num_nodes: The number of nodes to create in the graph
+/// :param float probability: The probability of an edge
+/// :param int seed: An optional seed to use for the random number generator
+///
+/// :return: A PyDiGraph object
+/// :rtype: PyDiGraph
+///
+/// .. [1] Vladimir Batagelj and Ulrik Brandes,
+///    "Efficient generation of large random networks",
+///    Phys. Rev. E, 71, 036113, 2005.
+/// .. [2] https://github.com/networkx/networkx/blob/networkx-2.4/networkx/generators/random_graphs.py#L49-L120
+#[pyfunction]
+#[text_signature = "(num_nodes, probability, seed=None, /)"]
+pub fn directed_gnp_random_graph(
+    py: Python,
+    num_nodes: isize,
+    probability: f64,
+    seed: Option<u64>,
+) -> PyResult<digraph::PyDiGraph> {
+    if num_nodes <= 0 {
+        return Err(IndexError::py_err("num_nodes must be > 0"));
+    }
+    let mut rng: Pcg64 = match seed {
+        Some(seed) => Pcg64::seed_from_u64(seed),
+        None => Pcg64::from_entropy(),
+    };
+    let mut inner_graph = StableDiGraph::<PyObject, PyObject>::new();
+    for x in 0..num_nodes {
+        inner_graph.add_node(x.to_object(py));
+    }
+    if probability <= 0.0 || probability >= 1.0 {
+        return Err(IndexError::py_err(
+            "Probability out of range, must be 0 < p < 1",
+        ));
+    }
+    let mut v: isize = 0;
+    let mut w: isize = -1;
+    let mut lp: f64 = 1.0 - probability;
+    lp = lp.ln();
+
+    while v < num_nodes {
+        let random: f64 = rng.gen_range(0.0, 1.0);
+        let mut lr = 1.0 - random;
+        lr = lr.ln();
```

Comment:
Is there a reason for having these in two separate lines instead of one? (Maybe this is just a rust convention)

Reply:
There was no real reason, I'll combine it 1 line.

Reply:
Done in: https://github.com/Qiskit/retworkx/pull/94/commits/25c0a1416e8fad86ba84207a62c77f289ef0245d